### PR TITLE
serialize:true takes priority over the OneToMany check for relationships

### LINKED ISSUE #3189: Serialize:true should take priority over the OneToMany check for relationships
https://github.com/emberjs/data/blob/ae6355af8235ca42d7dba0b3cf7a400732aa3c99/packages/ember-data/lib/serializers/json-serializer.js#L644 we should have a check for something like `mustSerialize` and account for it https://github.com/emberjs/data/blob/ae6355af8235ca42d7dba0b3cf7a400732aa3c99/packages/ember-data/lib/serializers/json-serializer.js#L656

### PR DESCRIPTION
**Pull request by sly7-7**
Fixes #3189 

Apologize in advance for my frenglished documentation
cc/ @igorT

### CODE REVIEW

File: packages/ember-data/lib/serializers/json-serializer.js [hunk 1]
```diff
@@ -111,6 +111,19 @@ export default Serializer.extend({
...
+    The `_mustSerialize` property is set to true when the user
+    explicitely use `{serialize: true}` when defining the attrs hash.
+
+    Specifically it takes priority over the other checks and the related
+    attribute/relationship will be serialized
+
+    @property _mustSerialize
+    @private
+  */
+  _mustSerialize: false,
```
Comment (igorT) [inline packages/ember-data/lib/serializers/json-serializer.js]: This doesn't make sense as a flag, it's a value that changes for each key, so needs to be a function, but we can just make it a part of canSerialize

**Comment by igorT** [comment]
@sly7-7 we should implement `shouldSerializeRelationship` and that can implement this logic + `oneToMany` check and then we call that instead of the check on 692 and the

**Comment by sly7-7** [comment]
@igorT I think I understand, I don't know when you want this, but I won't be able to do before tomorrow evening (trip today, soccer tomorrow). If it's not to late I'm happy to take that.

File: packages/ember-data/lib/serializers/json-serializer.js (hunk 1 above)
Comment (sly7-7) [inline packages/ember-data/lib/serializers/json-serializer.js]: Did'nt thought about that, definitely I will remove that.

**Comment by igorT** [comment]
No rush, enjoy the game

**Comment by sly7-7** [comment]
@igorT Ping ?

**Comment by igorT** [comment]
Seems to fail on Travis due to jshint

**Comment by sly7-7** [comment]
@igorT I'm confused, I can't reproduce the jshint error locally. I just rebase and force push... let's see whan happen

**Comment by sly7-7** [comment]
@igorT It's ready if you don't have any other (welcome) advices :)

**Comment by igorT** [comment]
@sly7-7 I think we talked about moving the 
```
+      if (this._shouldSerializeHasMany(key, relationshipType)) {
```
to be next to the canSerialize line right?

**Comment by igorT** [comment]
Thanks!

**Pull request merged**